### PR TITLE
Migration to algebra, redux

### PR DESCRIPTION
**Pull request by denisrosset**

### CODE REVIEW

**Comment by non** [comment]
Looks like you need to bump the algebra version to 0.6.0.

**Comment by denisrosset** [comment]
Yep! @non, can we chat briefly on gitter?

File: core/shared/src/main/scala/spire/math/Polynomial.scala [hunk 1]
```diff
@@ -400,15 +402,10 @@ trait Polynomial[@sp(Double) C] { lhs =>
...
   }
 
-  // EuclideanRing ops.
-
   def unary_-()(implicit ring: Rng[C]): Polynomial[C]
   def +(rhs: Polynomial[C])(implicit ring: Semiring[C], eq: Eq[C]): Polynomial[C]
   def -(rhs: Polynomial[C])(implicit ring: Rng[C], eq: Eq[C]): Polynomial[C] = lhs + (-rhs)
   def *(rhs: Polynomial[C])(implicit ring: Semiring[C], eq: Eq[C]): Polynomial[C]
-  def /~(rhs: Polynomial[C])(implicit field: Field[C], eq: Eq[C]): Polynomial[C] = (lhs /% rhs)._1
-  def /%(rhs: Polynomial[C])(implicit field: Field[C], eq: Eq[C]): (Polynomial[C], Polynomial[C])
```
Comment (denisrosset) [inline core/shared/src/main/scala/spire/math/Polynomial.scala]: Is there any implementation method I missed?

**Comment by denisrosset** [comment]
Leaving for dinner now. Let's see what CI has to say for 2.10.6 and for code coverage.

**Comment by denisrosset** [comment]
Indeed... great, more 2.10.6 specialization bugs on the plate. Will address that in 2-3 hours.

**Comment by denisrosset** [comment]
Help! something is wrong with the JS continuous integration. "Works for me" though on my laptop.

**Comment by denisrosset** [comment]
I'm +1 with merging if any other maintainer can fix the build.

**Comment by non** [comment]
@denisrosset Do you think this is ready to merge? If it's alright with you and @tixxit I'd like to merge this ASAP.

(Also thanks for fixing these issues!)

**Comment by tixxit** [comment]
👍

**Comment by denisrosset** [comment]
@non: ready to merge, yes. We need to update master such that the PRs are bite-sized, other nothing is going to get reviewed with our schedules!

Thanks to all!

**Pull request merged**